iam_groups_authn/sync.py

```diff
@@ -339,16 +326,20 @@ async def grant_iam_group_role(
         role: IAM group role.
         users_with_roles_future: Future for list of database users who have group role.
         iam_users_future: Future for list of IAM users in IAM group.
+        database_type: Type of database.
     """
     # await dependent tasks
     iam_users = await iam_users_future
     users_with_roles = await users_with_roles_future
 
-    # truncate mysql_usernames
-    mysql_usernames = [mysql_username(user) for user in iam_users]
+    if database_type == "mysql":
+        # truncate mysql_usernames
+        usernames = [mysql_username(user) for user in iam_users]
```

Comment:
just to double check - we can't push this sanitization into the function for `iam_futute`?

Reply:
Not really in an efficient manner, the `iam_future` is at the top of the task hierarchy and is the first task we schedule. So for it to know the `database_version` would require it to be scheduled later on and would impact the overall run time.